src/neo/Consensus/Commit.cs

```diff
@@ -5,22 +5,24 @@ namespace Neo.Consensus
 {
     public class Commit : ConsensusMessage
     {
-        public byte[] Signature;
-
-        public override int Size => base.Size + Signature.Length;
+        public byte[] BlockSignature;
+        public byte[] StateRootSignature;
```

Comment:
This should be the state root of the previous block.

Reply:
Put state root of previous block in `commit` and relay state root with one signature when collected enough `commit`?

Reply:
The state root of previous block may need to follow a full consensus agreement with `2F + 1`, perhaps we need to send that on `PrepResp`and `PrepReq`

Reply:
@erikzhang

Reply:
I think we can put the signatures of previous  state root into `PrepareRequest` and `PrepareResponse`. The state root can be calculated, we don't need to send it.

Reply:
Will do.

Reply:
#1719